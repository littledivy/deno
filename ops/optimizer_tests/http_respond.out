#[allow(non_camel_case_types)]
///Auto-generated by `deno_ops`, i.e: `#[op]`
///
///Use `op_flash_respond::decl()` to get an op-declaration
///you can include in a `deno_core::Extension`.
pub struct op_flash_respond;
#[doc(hidden)]
impl op_flash_respond {
    pub fn name() -> &'static str {
        stringify!(op_flash_respond)
    }
    pub fn v8_fn_ptr<'scope>() -> deno_core::v8::FunctionCallback {
        use deno_core::v8::MapFnTo;
        Self::v8_func.map_fn_to()
    }
    pub fn decl<'scope>() -> deno_core::OpDecl {
        deno_core::OpDecl {
            name: Self::name(),
            v8_fn_ptr: Self::v8_fn_ptr(),
            enabled: true,
            fast_fn: Some(
                Box::new(op_flash_respond_fast {
                    _phantom: ::std::marker::PhantomData,
                }),
            ),
            is_async: false,
            is_unstable: false,
            is_v8: false,
            argc: 4usize,
        }
    }
    #[inline]
    #[allow(clippy::too_many_arguments)]
    fn call(
        op_state: &mut OpState,
        server_id: u32,
        token: u32,
        response: &str,
        shutdown: bool,
    ) -> u32 {}
    pub fn v8_func<'scope>(
        scope: &mut deno_core::v8::HandleScope<'scope>,
        args: deno_core::v8::FunctionCallbackArguments,
        mut rv: deno_core::v8::ReturnValue,
    ) {
        let ctx = unsafe {
            &*(deno_core::v8::Local::<deno_core::v8::External>::cast(args.data()).value()
                as *const deno_core::_ops::OpCtx)
        };
        let arg_0 = args.get(0usize as i32);
        let arg_0 = match deno_core::serde_v8::from_v8(scope, arg_0) {
            Ok(v) => v,
            Err(err) => {
                let msg = format!(
                    "Error parsing args at position {}: {}", 0usize,
                    deno_core::anyhow::Error::from(err)
                );
                return deno_core::_ops::throw_type_error(scope, msg);
            }
        };
        let arg_1 = args.get(1usize as i32);
        let arg_1 = match deno_core::serde_v8::from_v8(scope, arg_1) {
            Ok(v) => v,
            Err(err) => {
                let msg = format!(
                    "Error parsing args at position {}: {}", 1usize,
                    deno_core::anyhow::Error::from(err)
                );
                return deno_core::_ops::throw_type_error(scope, msg);
            }
        };
        let arg_2 = match deno_core::v8::Local::<
            deno_core::v8::String,
        >::try_from(args.get(2usize as i32)) {
            Ok(v8_string) => deno_core::serde_v8::to_utf8(v8_string, scope),
            Err(_) => {
                return deno_core::_ops::throw_type_error(
                    scope,
                    format!("Expected string at position {}", 2usize),
                );
            }
        };
        let arg_2 = arg_2.as_ref();
        let arg_3 = args.get(3usize as i32);
        let arg_3 = match deno_core::serde_v8::from_v8(scope, arg_3) {
            Ok(v) => v,
            Err(err) => {
                let msg = format!(
                    "Error parsing args at position {}: {}", 3usize,
                    deno_core::anyhow::Error::from(err)
                );
                return deno_core::_ops::throw_type_error(scope, msg);
            }
        };
        let result = Self::call(
            &mut std::cell::RefCell::borrow_mut(&ctx.state),
            arg_0,
            arg_1,
            arg_2,
            arg_3,
        );
        let op_state = ::std::cell::RefCell::borrow(&*ctx.state);
        op_state.tracker.track_sync(ctx.id);
        match deno_core::serde_v8::to_v8(scope, result) {
            Ok(ret) => rv.set(ret),
            Err(err) => {
                deno_core::_ops::throw_type_error(
                    scope,
                    format!(
                        "Error serializing return: {}",
                        deno_core::anyhow::Error::from(err)
                    ),
                )
            }
        };
    }
}
struct op_flash_respond_fast {
    _phantom: ::std::marker::PhantomData<()>,
}
impl<'scope> deno_core::v8::fast_api::FastFunction for op_flash_respond_fast {
    fn function(&self) -> *const ::std::ffi::c_void {
        op_flash_respond_fast_fn as *const ::std::ffi::c_void
    }
    fn args(&self) -> &'static [deno_core::v8::fast_api::Type] {
        use deno_core::v8::fast_api::Type::*;
        use deno_core::v8::fast_api::CType;
        &[V8Value, Uint32, Uint32, SeqOneByteString, Bool, CallbackOptions]
    }
    fn return_type(&self) -> deno_core::v8::fast_api::CType {
        deno_core::v8::fast_api::CType::Uint32
    }
}
fn op_flash_respond_fast_fn<'scope>(
    _: deno_core::v8::Local<deno_core::v8::Object>,
    server_id: u32,
    token: u32,
    response: *const deno_core::v8::fast_api::FastApiOneByteString,
    shutdown: bool,
    fast_api_callback_options: *mut deno_core::v8::fast_api::FastApiCallbackOptions,
) -> u32 {
    use deno_core::v8;
    use deno_core::_ops;
    let __opts: &mut v8::fast_api::FastApiCallbackOptions = unsafe {
        &mut *fast_api_callback_options
    };
    let __ctx = unsafe {
        &*(v8::Local::<v8::External>::cast(unsafe { __opts.data.data }).value()
            as *const _ops::OpCtx)
    };
    let op_state = &mut ::std::cell::RefCell::borrow_mut(&__ctx.state);
    let response = unsafe { &*response }.as_str();
    let result = op_flash_respond::call(op_state, server_id, token, response, shutdown);
    result
}
