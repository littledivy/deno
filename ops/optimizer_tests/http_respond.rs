fn op_flash_respond(
  op_state: &mut OpState,
  server_id: u32,
  token: u32,
  response: &str,
  shutdown: bool,
) -> u32 {
}
