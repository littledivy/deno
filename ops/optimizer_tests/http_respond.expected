=== Optimizer Dump ===
returns_result: false
has_ref_opstate: true
has_rc_opstate: false
has_fast_callback_option: false
needs_fast_callback_option: false
fast_result: Some(U32)
fast_parameters: [V8Value, U32, U32, SeqOneByteString, Bool]
transforms: {3: Transform { kind: SeqOneByteString(Ref), index: 3 }}
is_async: false
fast_compatible: true
